next-tavla/app/(admin)/edit/[id]/components/TileCard/index.tsx

```diff
@@ -247,48 +261,77 @@ function TileCard({
-                                </Label>
+                        {!address && (
+                            <Label className="!text-error">
+                                {demoBoard
+                                    ? 'Logg inn for å få tilgang til funksjonaliteten.'
+                                    : 'Du må legge til en lokasjon for å kunne skru på gåavstand.'}
+                            </Label>
+                        )}
+                        <Switch
+                            name="showDistance"
+                            disabled={!address}
+                            defaultChecked={
+                                tile.walkingDistance?.visible ?? false
+                            }
+                        >
+                            Vis gåavstand
+                        </Switch>
+
+                        <Heading4>Forskyv avgangstid</Heading4>
+                        <div className="flex flex-col gap-2">
+                            <SubParagraph>
+                                Vis kun avganger som går om mer enn et valgt
+                                antall minutter.
+                            </SubParagraph>
+                            <TextField
+                                label="Antall minutter"
+                                name="offset"
+                                id="offset"
+                                type="number"
+                                min={0}
+                                className="!w-2/5"
+                                value={offset}
+                                onChange={(e) => {
+                                    isNaN(e.target.valueAsNumber)
+                                        ? setOffset('')
+                                        : setOffset(e.target.valueAsNumber)
+                                }}
+                                readOnly={offsetBasedOnWalkingDistance}
+                            />
+                            {address && (
+                                <Checkbox
+                                    checked={offsetBasedOnWalkingDistance}
+                                    onChange={() => {
+                                        setOffsetBasedOnWalkingDistance(() => {
+                                            !offsetBasedOnWalkingDistance
+                                                ? setOffset(
+                                                      walkingDistanceInMinutes,
+                                                  )
+                                                : setOffset(tile.offset ?? '')
+                                            return !offsetBasedOnWalkingDistance
+                                        })
+
+                                        posthog.capture(
+                                            'OFFSET_BASED_ON_WALKING_DISTANCE_BTN_CLICK',
+                                        )
+                                    }}
```

Comment:
Hvorfor må det defineres en egen funksjon for dette? (Funksjonen som er definert er heller ikke pure)

Kunne vi ikke bare satt offsetten direkte:

```tsx
if (offsetBasedOnWalkingDistance)
  setOffset(walkingDistanceInMinutes)
else
  setOffset(tile.offset)
setOffsetBasedOnWalkingDistance(!offsetBasedOnWalkingDistance)
posthog.capture('OFFSET_BASED_ON_WALKING_DISTANCE_BTN_CLICK')
```

Reply:
vi kan det, logikken blir da: `if (!offsetBasedOnWalkingDistance)...` fordi den ikke er oppdatert enda. Grunnen til at jeg gjorde det slik var fordi jeg ville "sikre meg" at jeg jobbet med de mest oppdaterte verdiene til `offsetBasedOnWalkingDistance` fordi vi skulle sette en annen state basert på denne. Vet ikke om det jeg har tenkt blir riktig da 🤔  https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state, som blir typ dette: 

```
onChange={() => {
    setOffsetBasedOnWalkingDistance(prevOffsetBasedOnWalkingDistance => {
        const newValue = !prevOffsetBasedOnWalkingDistance;

        if (newValue) {
            setOffset(walkingDistanceInMinutes);
        } else {
            setOffset(tile.offset ?? '');
        }

        return newValue;
    });
}}
```